patches/ui-webui-resources-tools-bundle_js_excludes.gni.patch

```diff
@@ -0,0 +1,12 @@
+diff --git a/ui/webui/resources/tools/bundle_js_excludes.gni b/ui/webui/resources/tools/bundle_js_excludes.gni
+index 2f9276480640dffdd5e73474ccab7d0e01259419..83e9c2547aa1946195d3165368e28f0193dff27e 100644
+--- a/ui/webui/resources/tools/bundle_js_excludes.gni
++++ b/ui/webui/resources/tools/bundle_js_excludes.gni
+@@ -13,6 +13,7 @@ foreach(excluded_file,
+           "resources/js/load_time_data.js",
+           "resources/ash/common/load_time_data.m.js",
+           "resources/mwc/lit/index.js",
++          "resources/brave/leo.bundle.js",
```

Comment:
Why is just this specific bundle a problem here?

Reply:
I think the best person to ask is @fallaciousreasoning, as this was at first introduced as a [patch](https://github.com/brave/brave-core/commit/62d74d54bbc9a7a040733bb7a78c9c11ab69cab4#diff-e53af1fe74e617071e19a23835fe42243871fc8951be9581e4f94b67fab4031c).

The patch used to have this comment:

```
# This variable comes from the upstream file, so disable the pylint warning
# about undefined variables.
# pylint: disable=undefined-variable
```

I guess I could add this to the commit description. It has no description because it was left on WIP for a while. The patch and the override have been deleted, as upstream has moved this mechanism into `gn`.

Reply:
what exactly is the error?

Reply:
```
[73/723] ACTION //chrome/browser/resources/settings:build_bundle(//build/toolchain/linux:clang_x64)
FAILED: gen/chrome/browser/resources/settings/bundled/settings.rollup.js gen/chrome/browser/resources/settings/bundled/settings.rollup.js.map gen/chrome/browser/resources/settings/bundled/lazy_load.rollup.js gen/chrome/browser/resources/settings/bundled/lazy_load.rollup.js.map gen/chrome/browser/resources/settings/bundled/shared.rollup.js gen/chrome/browser/resources/settings/bundled/shared.rollup.js.map gen/chrome/browser/resources/settings/bundled/rollup.config.mjs gen/chrome/browser/resources/settings/bundled/build_bundle_requestlist.txt
python3 ../../ui/webui/resources/tools/bundle_js.py --host settings --input gen/chrome/browser/resources/settings/tsc --out_folder gen/chrome/browser/resources/settings/bundled --depfile gen/chrome/browser/resources/settings/build_bundle.d --target_name build_bundle --exclude chrome://resources/mojo/mojo/public/js/bindings.js //resources/mojo/mojo/public/js/bindings.js chrome://resources/mojo/mojo/public/mojom/base/time.mojom-lite.js //resources/mojo/mojo/public/mojom/base/time.mojom-lite.js chrome://resources/polymer/v3_0/polymer/polymer_bundled.min.js //resources/polymer/v3_0/polymer/polymer_bundled.min.js chrome://resources/js/cr.js //resources/js/cr.js chrome://resources/js/load_time_data.js //resources/js/load_time_data.js chrome://resources/ash/common/load_time_data.m.js //resources/ash/common/load_time_data.m.js chrome://resources/mwc/lit/index.js //resources/mwc/lit/index.js chrome://resources/mojo/skia/public/mojom/skcolor.mojom-webui.js --external_paths /shared/settings\|gen/chrome/browser/resources/settings_shared/tsc chrome://resources/polymer/v3_0/\|../../third_party/polymer/v3_0/components-chromium/ //resources/polymer/v3_0/\|../../third_party/polymer/v3_0/components-chromium/ chrome://resources/brave/\|gen/brave/ui/webui/resources/tsc/ //resources/brave/\|gen/brave/ui/webui/resources/tsc/ chrome://resources/\|gen/ui/webui/resources/tsc/ //resources/\|gen/ui/webui/resources/tsc/ --js_module_in_files settings.js lazy_load.js
Traceback (most recent call last):
  File "../../ui/webui/resources/tools/bundle_js.py", line 277, in <module>
    main(sys.argv[1:])
  File "../../ui/webui/resources/tools/bundle_js.py", line 250, in main
    optimize_output = _optimize(args.input, args)
  File "../../ui/webui/resources/tools/bundle_js.py", line 217, in _optimize
    js_module_out_files = _bundle(out_path, in_path, manifest_out_path, args,
  File "../../ui/webui/resources/tools/bundle_js.py", line 171, in _bundle
    node.RunNode([node_modules.PathToRollup()] + rollup_args + [
  File "/home/cdesouza/dev/brave-browser/src/out/Component/../../third_party/node/node.py", line 34, in RunNode
    raise RuntimeError('Command \'%s\' failed\n%s' % (' '.join(cmd), err))
RuntimeError: Command '/home/cdesouza/dev/brave-browser/src/out/Component/../../third_party/node/linux/node-linux-x64/bin/node /home/cdesouza/dev/brave-browser/src/out/Component/../../third_party/node/node_modules/rollup/dist/bin/rollup /home/cdesouza/dev/brave-browser/src/out/Component/gen/chrome/browser/resources/settings/tsc/settings.js /home/cdesouza/dev/brave-browser/src/out/Component/gen/chrome/browser/resources/settings/tsc/lazy_load.js --chunkFileNames shared.rollup.js --format esm --dir /home/cdesouza/dev/brave-browser/src/out/Component/gen/chrome/browser/resources/settings/bundled --entryFileNames [name].rollup.js --sourcemap --sourcemapExcludeSources --config /home/cdesouza/dev/brave-browser/src/out/Component/gen/chrome/browser/resources/settings/bundled/rollup.config.mjs' failed

/home/cdesouza/dev/brave-browser/src/out/Component/gen/chrome/browser/resources/settings/tsc/settings.js, /home/cdesouza/dev/brave-browser/src/out/Component/gen/chrome/browser/resources/settings/tsc/lazy_load.js → gen/chrome/browser/resources/settings/bundled...
[!] Error: Could not load gen/brave/ui/webui/resources/tsc/leo.bundle.js (imported by gen/chrome/browser/resources/settings/tsc/brave_overrides/iron_icon.js): ENOENT: no such file or directory, open 'gen/brave/ui/webui/resources/tsc/leo.bundle.js'
Error: Could not load gen/brave/ui/webui/resources/tsc/leo.bundle.js (imported by gen/chrome/browser/resources/settings/tsc/brave_overrides/iron_icon.js): ENOENT: no such file or directory, open 'gen/brave/ui/webui/resources/tsc/leo.bundle.js'
```

Reply:
So this JS bundle is shared across a number of WebUI pages. Chromium duplicates shared code into each bundle which uses that code unless it's added to this exclude list.

Essentially, this is saying "load at runtime from chrome://resources" rather than "include this code in our bundle"

The error appears without it because it's trying to locate the bundle in the path mapping for `chrome://resources/brave` (so it can be added to the current bundle), which is not where the build output from Leo (Nala) goes.

Reply:
Fixed.